test/registry.spec.js

```diff
@@ -1,183 +1,303 @@
-describe('Registry', function () {
+describe('Registry', function() {
 
-  describe('constructor', function () {
-
-    it('should fail if not called with new', function () {
-      expect(Registry).to.throwException();
-    });
-
-    it('should fail if not called with a loader object', function() {
-      expect(function() { new Registry(undefined) }).to.throwException();
-      expect(function() { new Registry('string') }).to.throwException();
-      expect(function() { new Registry(0) }).to.throwException();
-      expect(function() { new Registry(function(){}) }).to.throwException();
-    });
-
-    it('should succeed if called with a loader object', function() {
-      new Registry({});
-    });
+  var moduleName = 'm1';
+  var moduleValue = { a: 'b', c: 'd' };
 
+  beforeEach(function() {
+    System.registry.set(moduleName, moduleValue);
   });
 
-  describe('polyfilled iteration of instances', function() {
+  afterEach(function() {
+    System.registry.delete(moduleName, moduleValue);
+  });
 
-    it('should iterate over the registryData', function () {
-      var registry = new Registry({});
-      registry._registry.registryData.push({ key: 'module1', entry: {} });
-      registry._registry.registryData.push({ key: 'module2', entry: {} });
+  describe('constructor', function() {
 
-      var iter = registry[Symbol.iterator]();
-      var index = 0;
-      var next;
-      while ( !(next = iter.next()).done ) {
-        expect(next.value).to.be(registry._registry.registryData[index++]);
-      }
+    //The constructor always throws because registries cannot be created in user-land
 
-      expect(index).to.be(registry._registry.registryData.length - 1);
+    it('should fail if not called with new', function() {
+      expect(Registry).to.throwException();
+    });
 
+    it('should fail if called with new', function() {
+      expect(function() { new Registry() }).to.throwException();
     });
 
   });
 
-  describe('install function', function() {
+  describeIf(System.transpiler !== 'traceur', 'entries function', function() {
```

Comment:
See https://travis-ci.org/ModuleLoader/es6-module-loader/builds/101356889 and https://travis-ci.org/ModuleLoader/es6-module-loader/jobs/101356890#L969